src/librustc/ty/mod.rs

```diff
@@ -2762,3 +2762,51 @@ impl fmt::Display for SymbolName {
         fmt::Display::fmt(&self.name, fmt)
     }
 }
+
+pub fn ty_fn_sig<'a, 'tcx: 'a>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
```

Comment:
Maybe combine this with the existing `fn_sig` helper?

Reply:
I just moved it from `librustc_trans/common.rs`. It also uses `fn_sig` internally.

Reply:
What I meant is that maybe we should make it part of `fn_sig` instead, so that you just write `tcx.fn_sig(...)`.